hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -1212,4 +1246,43 @@ private void completeObservables(Set<String> observables, Throwable error) {
             return lmc == null || lmc instanceof LightMasterContext && ((LightMasterContext) lmc).isCancelled();
         }).toArray();
     }
+
+    public void scheduleJobTimeout(final long jobId, final long timeout, final boolean lightJob) {
+        if (!lightJob && !isMaster()) {
+            return;
+        }
+
+        if (scheduledJobTimeouts.containsKey(jobId)) {
+            return;
+        }
+
+        scheduledJobTimeouts.computeIfAbsent(jobId, id -> scheduleJobTimeoutTask(id, timeout));
```

Comment:
I thought about removing `!lightJob && !isMaster()` too, but I think there's a split-brain scenario in which it would prevent "double cancellation" (not that it would be much of a problem though I suspect). Specifically: 
1. Node-1 (original master) schedules timeout task. 
2. Node-1 is disconnected from the rest of the cluster. 
3. Node-2 is elected new master and schedules another timeout task on itself. 
4. Node-1 rejoins cluster. 
5. Job times out - both Node-1 and Node-2 try to terminate it - not sure how undesirable this outcome is. 

As far as performing all the checks here actually, and possibly renaming this function into `tryScheduleTimeout`, this could work, but I'm not sure how expressive it would be. I'll look into it.